src/status_im/anon_metrics/core.cljs

```diff
@@ -0,0 +1,90 @@
+(ns status-im.anon-metrics.core
+  (:require [status-im.ethereum.json-rpc :as json-rpc]
+            [taoensso.timbre :as log]
+            [re-frame.core :as re-frame]
+            [re-frame.interceptor :refer [->interceptor]]
+            [status-im.utils.async :refer [async-periodic-exec async-periodic-stop!]]
+            [status-im.utils.platform :as platform]
+            [status-im.utils.build :as build]
+            [status-im.utils.fx :as fx]
+            [status-im.anon-metrics.transformers :as txf]))
+
+(defonce events-foyer (atom []))
+(defonce periodic-tasks-chan (atom nil))
+
+(defn onboard-events
+  "Check if there are any events in the foyer, flush them to the backend and clear foyer on-success."
+  []
+  (let [outstanding-events @events-foyer]
+    (when (seq outstanding-events)
+      (reset! events-foyer [])
+      (json-rpc/call {:method     "appmetrics_saveAppMetrics"
+                      :params     [outstanding-events]
+                      :on-error   (fn [err]
+                                    (log/error {:error err
+                                                :events outstanding-events})
+                                    (log/warn "All outstanding events will be rejected"))}))))
+
+(re-frame/reg-fx
+ ::transfer-data
+ (fn [transfer?]
+   (if (and transfer?
+            ;; double run safety
+            (not @periodic-tasks-chan))
+     ;; run the onboard-events fn periodically
+     (reset! periodic-tasks-chan
+             ;; interval = 4000 ms (run every `interval` milliseconds)
+             ;; timeout = 5000 ms (exit if the fn doesn't exit within `timeout` milliseconds)
+             (async-periodic-exec onboard-events 4000 5000))
+     (async-periodic-stop! @periodic-tasks-chan))))
```

Comment:
will periodic-tasks-chan be cleared here ?

Reply:
```clojure
(defn async-periodic-stop! [async-periodic-chan]
  (async/close! async-periodic-chan))
```

`async/close!` will will only close the channel but not clear it. We don't have anything in the channel, just a timeout. Everything is stored in the `events-foyer` atom. So we should clear that manually.

Reply:
yes thats why im asking, i don't see we clear it manually

Reply:
I called the `onboard-events` function, that will clear the atom and also save everything to the db.

Reply:
mhm i don't see if it cleans periodic-tasks-chan

Reply:
We don't put anything on the channel. It's never "dirty" for us to "clean".
https://github.com/status-im/status-react/blob/90ad5f2f77964eff1a7b155e475dea79f5ae4622/src/status_im/utils/async.cljs#L67